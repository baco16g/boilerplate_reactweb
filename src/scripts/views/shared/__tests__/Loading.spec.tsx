import * as React from 'react';
import {render, cleanup} from 'react-testing-library';
import 'jest-dom/extend-expect';
import 'jest-styled-components';
import Loading from '../Loading';

afterEach(cleanup);

it('matches snapshot', () => {
  const {asFragment} = render(<Loading />);
  expect(asFragment()).toMatchSnapshot();
});

it('The style passed in props should be applied correctly', () => {
  const {container} = render(<Loading width={100} height={100} />);
  expect(container.firstChild).toHaveStyleRule('width', '100px');
  expect(container.firstChild).toHaveStyleRule('height', '100px');
});
