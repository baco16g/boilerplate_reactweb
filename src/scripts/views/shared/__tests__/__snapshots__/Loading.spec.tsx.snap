// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`matches snapshot 1`] = `
<DocumentFragment>
  .c0 {
  position: absolute;
  width: 64px;
  height: 64px;
  top: 0;
  left: 0;
  right: 0;
  bottom: 0;
  margin: auto;
}

<div
    class="c0"
    height="64"
    width="64"
  >
    <div
      style="fill: #fff; height: 64px; width: 64px;"
    >
      <svg
        class="icon-blank"
        viewBox="0 0 32 32"
        xmlns="http://www.w3.org/2000/svg"
      />
      

    </div>
  </div>
</DocumentFragment>
`;
